addons/google_recaptcha/static/src/js/signup.js

```diff
@@ -0,0 +1,43 @@
+/** @odoo-module **/
+
+import publicWidget from "@web/legacy/js/public/public_widget";
+import { ReCaptcha } from "@google_recaptcha/js/recaptcha";
+
+const CaptchaFunctionality = {
+    events: {
+        submit: "_onSubmit",
+    },
+
+    init() {
+        this._super(...arguments);
+        this._recaptcha = new ReCaptcha();
+    },
+
+    async willStart() {
+        return this._recaptcha.loadLibs();
+    },
+
+    _onSubmit(event) {
+        if (!this.$el.find("input[name='recaptcha_token_response']").length) {
+            event.preventDefault();
+            this._recaptcha.getToken(this.token_name).then((tokenCaptcha) => {
+                this.$el.append(
+                    `<input name="recaptcha_token_response" type="hidden" value="${tokenCaptcha.token}"/>`,
+                );
+                this.$el.submit();
```

Comment:
True for the whole PR: we normally don't add *new* jQuery code anywhere. But since it's already merged in master, let's not care.